plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -177,26 +192,147 @@ func (s *SpanReader) FindTraces(queryParams *spanstore.TraceQueryParameters) ([]
 	return retMe, nil
 }
 
+// FindTraces queries for traces.
+func (s *SpanReader) getTraceIDs(traceQuery *spanstore.TraceQueryParameters) ([]dbmodel.TraceID, error) {
+	if traceQuery.DurationMin != 0 {
+		return s.queryTraceIdsByDuration(traceQuery)
+	}
+
+	// NB(rooz): consider refactoring the query... methods where they take in traceQuery instead of a breakdown of its values
+	if traceQuery.OperationName != "" {
+		traceIds, err := s.queryByServiceNameAndOperation(traceQuery.ServiceName, traceQuery.OperationName, traceQuery.StartTimeMin, traceQuery.StartTimeMax, traceQuery.NumTraces)
+		if err != nil {
+			return nil, err
+		}
+		if len(traceQuery.Tags) > 0 {
+			tagTraceIds, err := s.queryByTagsAndLogs(traceQuery.ServiceName, traceQuery.StartTimeMin, traceQuery.StartTimeMax, traceQuery.Tags, traceQuery.NumTraces)
+			if err != nil {
+				return nil, err
+			}
+			return s.intersectTraceIDSlices(traceIds, tagTraceIds), nil
+		}
+		return traceIds, err
+	}
+	if len(traceQuery.Tags) > 0 {
+		return s.queryByTagsAndLogs(traceQuery.ServiceName, traceQuery.StartTimeMin, traceQuery.StartTimeMax, traceQuery.Tags, traceQuery.NumTraces)
+	}
+	return s.queryTraceIdsByService(traceQuery.ServiceName, traceQuery.StartTimeMin, traceQuery.StartTimeMax, traceQuery.NumTraces)
+}
+
+func (s *SpanReader) intersectTraceIDSlices(one, other []dbmodel.TraceID) []dbmodel.TraceID {
```

Comment:
pretty sure we already had a function like this before when we were intersecting main and tags query results